docs/tutorial/data-flow.md

```diff
@@ -0,0 +1,153 @@
+## Data Flow
+
+There are several methods for sending data between controllers and components. This tutorial provides an overview of the different possibilities and best practices for your application.
```

Comment:
I'd like to add a simple decision tree that links to the relevant heading depending on choices. E.g

```md

- Do you need bidirectional data flow?
  - Yes → use [Properties](#properties)
- Do you want to pass data from child to parent only?
  - Use [Consumer](#consumer)
- Do you want to update the child at some later point after initialization?
  - Use Java setters
- Otherwise, use regular Params
```